src/test/java/com/dragonballs/services/user/UserServiceTest.java

```diff
@@ -0,0 +1,230 @@
+package com.dragonballs.services.user;
+
+import com.dragonballs.dao.UserDAO;
+import com.dragonballs.entities.User;
+import com.dragonballs.exceptions.*;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.*;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserServiceTest {
```

Comment:
Multiple repeats! Of mocking, test data, data creation - no duplications allowed